node-hub/ultralytics-yolo/main.py

```diff
@@ -0,0 +1,72 @@
+import os
+import argparse
+
+import numpy as np
+import pyarrow as pa
+
+from dora import Node
+from ultralytics import YOLO
+
+def main():
+    # Handle dynamic nodes, ask for the name of the node in the dataflow, and the same values as the ENV variables.
+    parser = argparse.ArgumentParser(
+        description="UltraLytics YOLO: This node is used to perform object detection using the UltraLytics YOLO model.")
+
+    parser.add_argument("--name", type=str, required=False, help="The name of the node in the dataflow.",
+                        default="ultralytics-yolo")
+    parser.add_argument("--model", type=str, required=False,
+                        help="The name of the model file (e.g. yolov8n.pt).", default="yolov8n.pt")
+
+    args = parser.parse_args()
+
+    model_path = os.getenv("MODEL", args.model)
+
+    model = YOLO(model_path)
+    node = Node(args.name)
+
+    pa.array([])  # initialize pyarrow array
+
+    for event in node:
+        event_type = event["type"]
+
+        if event_type == "INPUT":
+            event_id = event["id"]
+
+            if event_id == "image":
+                arrow_image = event["value"][0]
+                image = {
+                    "width": np.uint32(arrow_image["width"].as_py()),
+                    "height": np.uint32(arrow_image["height"].as_py()),
+                    "channels": np.uint8(arrow_image["channels"].as_py()),
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                }
+
+                frame = image["data"].reshape((image["height"], image["width"], 3))
+
+                frame = frame[:, :, ::-1]  # OpenCV image (BGR to RGB)
```

Comment:
I wonder if instead of channel being a uint8, if we should instead use an encoding string of type: `RGB` or `BGR` or else as I can see this as easily error prone.

I wonder if we should just copy ros2 message format: 
https://docs.ros.org/en/noetic/api/sensor_msgs/html/msg/Image.html

I think the encoding definition is still interesting:

https://docs.ros.org/en/indigo/api/sensor_msgs/html/image__encodings_8h_source.html